third_party/proton/proton/hooks/instrumentation.py

```diff
@@ -61,22 +60,61 @@ def load_dialects(self, ctx):
         triton_proton.load_dialects(ctx)
 
 
+def _interpret_mode(mode_obj: Union[str, mode.InstrumentationMode]) -> mode.InstrumentationMode:
+    if isinstance(mode_obj, mode.InstrumentationMode):
+        return mode_obj
+
+    parts = mode_obj.split(":")
+    mode_name = parts[0]
+    opts = dict(opt.split("=") for opt in parts[1:])
+
+    # Get option values or empty strings
+    options = {
+        "metric_type": opts.get("metric_type", "cycle"),
+        "buffer_type": opts.get("buffer_type", "shared"),
+        "buffer_strategy": opts.get("buffer_strategy", "circular"),
+        "buffer_size": int(opts.get("buffer_size", "0")),
+        "granularity": opts.get("granularity", "warp"),
+        "sampling_strategy": opts.get("sampling_strategy", "none"),
+        "sampling_options": opts.get("sampling_options", ""),
+    }
```

Comment:
How are these options actually set in front end? Is it something like:

proton.start(filename, backend="instrumentation", mode="buffer_type = shared")?

Reply:
That's a good question. Yeah there will be two ways to pass the mode:

1. "mode_name:metric_type =cycle:buffer_type=shared:buffer_size=1024"
2. Initialize a mode object. See examples in `mode.py`

Reply:
I see. Could we will fill in one of those methods just using a couple of the default values in test_instrumentation.py below? It should be a NFC to the test itself if we just use the existing default values

Reply:
>  Could we fill in one of those methods just using one or two of the default values (e.g. just buffer_type=shared for example and can leave the rest off to just revert to their defaults)

Yes, I guess you just wanted to test the feature? In fact, I think I should refactor `test_instrumentation.py` a bit as well to decouple features. 

Please stay tuned a bit until the end of this week maybe, I will be traveling from tomorrow until Friday

Reply:
Yeah, well two things:

1. A sanity check that the defaults should be a NFC to the existing test. 
2. I want get https://github.com/triton-lang/triton/pull/6449/files merged after this PR and just want to make sure I understand the options are intended to be set so I can do the stack buffer version of the test.